src/ribasim_nl/ribasim_nl/cloud.py

```diff
@@ -0,0 +1,282 @@
+import logging
+import os
+import shutil
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, Tuple, Union
+from xml.etree import ElementTree
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+RIBASIM_NL_CLOUD_PASS = os.getenv("RIBASIM_NL_CLOUD_PASS")
+RIBASIM_NL_DATA_DIR = os.getenv("RIBASIM_NL_DATA_DIR")
+RIBASIM_NL_CLOUD_USER = "nhi_api"
+WEBDAV_URL = "https://deltares.thegood.cloud/remote.php/dav"
+BASE_URL = rf"{WEBDAV_URL}/files/{RIBASIM_NL_CLOUD_USER}/Ribasim modeldata"
+
+WATER_AUTHORITIES = [
+    "AaenMaas",
+    "AmstelGooienVecht",
+    "BrabantseDelta",
+    "DeDommel",
+    "Delfland",
+    "DrentsOverijsselseDelta",
+    "HollandseDelta",
+    "HollandsNoorderkwartier",
+    "HunzeenAas",
+    "Limburg",
+    "Noorderzijlvest",
+    "RijnenIJssel",
+    "Rijnland",
+    "Rivierenland",
+    "Rijkswaterstaat",
+    "Scheldestromen",
+    "SchielandendeKrimpenerwaard",
+    "StichtseRijnlanden",
+    "ValleienVeluwe",
+    "Vechtstromen",
+    "WetterskipFryslan",
+    "Zuiderzeeland",
+]  # noqa
+
+HIDDEN_DIRS = ["D-HYDRO modeldata"]  # somehow this dir-name still exists :-(
+
+
+def is_dir(item):
+    return Path(item).suffix == ""
+
+
+@dataclass
+class Cloud:
+    data_dir: Union[str, Path] = RIBASIM_NL_DATA_DIR
```

Comment:
I see you convert to Path in `__post_init__`. Can't we not drop the Union here and make sure it is Path from the start?

I prefer to be less flexible to keep this code as simple as possible.

Reply:
1. From the user-perspective; most readers accept Path and str-type paths; e.g. gpd.read_file(Path("my_file.shp")) and gpd.read_file("my_file.shp") both work.   
2. It will be a string (or None) if users do or do not supply `RIBASIM_NL_DATA_DIR` or the data_dir at `__init__` https://github.com/Deltares/Ribasim-NL/blob/a3aa8b8f55fac0276372cd9546c38799e37368a1/src/ribasim_nl/ribasim_nl/cloud.py#L76

Reply:
Yeah I'm not saying we should not accept strings from the user, just that in our class it seemed like it would be easier if we convert it at the door to this class. But I don't directly have a suggestion how to do this, so feel free to ignore.